lib/chat_web/live/main_live/page/lobby.ex

```diff
@@ -91,24 +90,57 @@ defmodule ChatWeb.MainLive.Page.Lobby do
       PubSub.broadcast!(
         Chat.PubSub,
         @topic,
-        :room_request
+        {:room_request, room_hash, me |> Identity.pub_key() |> Utils.hash()}
       )
     end)
 
     socket
     |> assign_room_list()
   end
 
-  def approve_requests(socket) do
+  def approve_room_request(%{assigns: %{room_map: room_map}} = socket, room_hash, user_hash)
+      when is_map_key(room_map, room_hash) do
+    {_, _, {encrypted, blob}} =
+      room_hash
+      |> Rooms.approve_requests(room_map[room_hash])
```

Comment:
How the room knows that `user_hash` made a request?

Reply:
done